core/src/main/java/org/elasticsearch/action/support/TransportAction.java

```diff
@@ -133,6 +133,10 @@ public final void execute(Task task, Request request, ActionListener<Response> l
             return;
         }
 
+        if (task != null && request.getShouldPersistResult()) {
+            listener = new PersistentActionListener<>(taskManager, task, listener);
```

Comment:
`newResponse` is not defined on the `TransportAction` level, it is only defined for some specific Actions such as `TransportMasterNodeAction`. So, I don't think I can really implement this assert here.